balrogscript/src/balrogscript/submitter/cli.py

```diff
@@ -30,6 +30,12 @@ def _getFileUrls(self, productName, version, buildNumber, updateChannels, ftpSer
         file_prefix = productName.lower()
         if file_prefix == "devedition":
             file_prefix = "firefox"
+        # bug 1444406 - eventually we're going to default to https
+        protocol = "http"
+        if productName.lower() == "devedition":
+            protocol = "https"
+        if productName.lower() == "firefox" and all([c.startswith(("beta",)) for c in updateChannels]):
```

Comment:
We spoke on video, so to record and correct what I said ... the idea is to restrict to beta for now, and later extend the tuple in startswith to add release, and maybe esr later still (or maybe do it all in one go, TBC). 

For the RC case, the main thing is to preserve http for release channels, and if we lose https we've had for the rest of the beta cycle then OK, that should only be temporary. But in fact we have two toplevel submit tasks for RCs, and the `release-secondary-balrog-submit-toplevel` kind (for RC-on-beta) will end up traversing the changes here too. I've checked in staging and we would get https for the beta channels, but there's a problem. The ordering of the two tasks matters because the last task overwrites the "*" match for completes - we actually want the secondary to run first, then the 'primary' to come along an set http for "*". Turns out the task order is not stable because there is no dependency relationship between them. I'll see if I can come up with a solution.

Reply:
5750195 resolves this by only writing the `"*"` in the main toplevel task, and secondary adds beta/beta-cdntest/beta-localtest. @mozbhearsum could you double-check the logic and test case ?